pkg/gatherers/clusterconfig/machines_test.go

```diff
@@ -0,0 +1,51 @@
+package clusterconfig
+
+import (
+	"context"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	dynamicfake "k8s.io/client-go/dynamic/fake"
+)
+
+func Test_Machine_Gather(t *testing.T) {
```

Comment:
I'm not a big fan of this kind of test, 'cause it doesn't actually cover all the possible scenarios. Consider using the Table Driven Test approach.

Reply:
☝️ this is just a suggestion.